modules/installation-user-infra-machines-advanced.adoc

```diff
@@ -176,127 +169,113 @@ $ ls $HOME/clusterconfig/
-+
-For PXE or ISO boots, you can create the Ignition config
-and APPEND the `ignition.config.url=` option to identify the location of
-the Ignition config. You also need to append `ignition.firstboot ignition.platform.id=metal`
-or the `ignition.config.url` option will be ignored.
+When doing an {op-system} bare metal installation, there are two types of Ignition configs that you can provide, with different reasons for providing each one:
 
 * **Permanent install Ignition config**: Every bare metal {op-system-first} installation
 needs to pass one of the Ignition config files generated by `openshift-installer`,
-including `bootstrap.ign`, `master.ign` and `worker.ign`, to the carry out the
+such as `bootstrap.ign`, `master.ign` and `worker.ign`, to carry out the
 installation. It is not recommended to modify these files.
 +
-For PXE installs, you pass the Ignition configs on the APPEND line using the
+For PXE installs, you pass the Ignition configs on the `APPEND` line using the
 `coreos.inst.ignition_url=` option. For ISO installs, after the ISO boots to
 the shell prompt, you identify the Ignition config on the `coreos-installer`
 command line with the `--ignition-url=` option.
 +
-Instead of passing the location of an Ignition config via a kernel or
+Instead of passing the location of an Ignition config using a kernel or
 command-line option, you can embed an Ignition config into the ISO
 installer image. This allows you to do bare metal installs with the ISO,
 without requiring access to an HTTP server. See “Embedding an Ignition
 config in the {op-system-first} ISO” for details.
 
+* **Live install Ignition config**: This type must be created manually and should be avoided if possible, as it is not supported by Red Hat. With this method, the Ignition config passes to the live install medium, runs immediately upon booting, and performs setup tasks before and/or after the {op-system} system installs to disk. This method should only be used for performing tasks that must be performed once and not applied again later, such as with advanced partitioning that cannot be done using a MachineConfig.
++
+For PXE or ISO boots, you can create the Ignition config
+and `APPEND` the `ignition.config.url=` option to identify the location of
+the Ignition config. You also need to append `ignition.firstboot ignition.platform.id=metal`
+or the `ignition.config.url` option will be ignored.
+
 [id="installation-user-infra-machines-advanced_embedignition_{context}"]
 === Embedding an Ignition config in the {op-system-first} ISO
-The following procedures describe how to embed an Ignition config into
-the ISO so it is applied when the new installation first boots from disk.
+To embed an Ignition config named `worker.ign` into an ISO image, for example `rhcos-<version>-live.x86_64.iso`, you must:
```

Comment:
We're still not converging on this one, so lemme try again:

Using `coreos-installer iso ignition embed` to embed a `worker.ign` is basically not feasible in 4.6.  The user would have to write a "live install Ignition config" that writes out the `worker.ign` as a file and then overrides the `coreos-installer.service` unit to read that file using `--ignition-file`.  This flow is well outside what we want users to do, especially since the doc just explained that live install Ignition configs are unsupported.  This means the `iso ignition embed` flow is not useful in any supported flow in 4.6.  I think there's value in documenting it anyway, since people will still use it, but we'll need to frame the instructions appropriately.

FYI, there's discussion of closing this gap for future releases; see https://github.com/openshift/enhancements/pull/492.

Reply:
Thanks for sticking with this. I went back and reread old threads from you and Chris. At first blush, my thought is that we should not be documenting something so far from any workflow in 4.6. But I can see the argument that it's better for customers that we at least mention it, since some will try, and then that lets us warn that it's not supported or recommended...and save potential support tickets.

As for practical application in the docs, I'm not sure that this latest revision of the procedure is getting it quite right, and I don't have the experience to test it myself. We're technically past doc freeze, so if there are any edits, please be very specific so I can be sure to get it right. Thanks!